rotkehlchen/chain/evm/decoding/aave/common.py

```diff
@@ -171,36 +204,75 @@ def _decode_deposit(
                 ):
                     event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
                     resolved_asset = event.asset.resolve_to_asset_with_symbol()
-                    event.notes = f'Receive {event.balance.amount} {resolved_asset.symbol} from {self.label}'  # noqa: E501
+                    event.balance.amount = amount
+                    event.notes = f'Receive {amount} {resolved_asset.symbol} from {self.label}'
                     event.counterparty = self.counterparty
                     receive_event = event
 
-        return deposit_event, receive_event
+        return deposit_event, receive_event, earned_event
 
     def _decode_withdrawal(
             self,
             token: 'EvmToken',
-            tx_log: EvmTxReceiptLog,
-            decoded_events: list['EvmEvent'],
-    ) -> tuple[EvmEvent | None, EvmEvent | None]:
+            context: 'DecoderContext',
+    ) -> tuple[EvmEvent | None, EvmEvent | None, EvmEvent | None]:
```

Comment:
have you checked if the logic works the same in v2 and v3? From the chat in tg I kind of recall that it was the same

Reply:
they seemingly have a different mechanism, but this logic doesn't interfere with them because the MINT and BURN topic is different for them

Reply:
I also asked this in another part of the PR. If this is only for v3 it should not be in the common logic. It goes against the expectations